README.md

```diff
@@ -1,41 +1,124 @@
 
-When reviewing the REP, the reviewers and the shepherd should apply the following judgements:
-- If an author proposes a change to be within the `ray` repo, the reviewers and the shepherd should assess whether the change can be layered on top of `ray` instead. 
-If so we should try to make the change in a separate repo. 
-- For a change proposed as an ecosystem project under `ray-project`: the reviewers and the shepherd should make sure that the technical quality
-meets the bar of (at least) a good "experimental" or "alpha" feature -- we should be comfortable welcoming Ray users with similar use cases to try this project.
-- For a change proposed as a new ecosystem project (outside of `ray-project`): then this REP is just serving as a "request for comments". 
-We don't need to go through the voting process, since it's not Ray committers' decision to approve the change. 
+### Should this change be within `ray` or outside?
+Yes. The change should be within `ray`.
 
 ## Stewardship
 ### Required Reviewers
-The proposal will be open to the public, but please suggest a few experienced Ray contributors in this technical domain whose comments will help this proposal. Ideally, the list should include Ray committers. 
+@ericl, @scv119
+
 ### Shepherd of the Proposal (should be a senior committer)
-To make the review process more productive, the owner of each proposal should identify a **shepherd** (should be a senior Ray committer). The shepherd is responsible for working with the owner and making sure the proposal is in good shape (with necessary information) before marking it as ready for broader review.
 
 ## Design and Architecture
-The proposal should include sufficient technical details for reviewers to determine the anticipated benefits and risks.
+The design consists of the object store plugin interface and the plugin manager.
+
+### Plugin Interface
+
+The plugin provides the functionalities of an object store responsible for storing and retrieving logic. It can be implemented as source code within the Ray repository or a shared library to be loaded dynamically. 
+
+It shall implement classes inherited from `ObjectStoreClientInterface` and `ObjectStoreRunnerInterface`.   
+`ObjectStoreClientInterface` replaces the current `PlasmaClientInterface`. It keeps all the functions from the current `PlasmaClientInterface` to retain full compatibility along with some extensions. Current `PlasmaClient` shall implement `ObjectStoreClientInterface`.
+
+New virtual functions of `ObjectStoreClientInterface` :
+```c++
+/// Authentication to the object store.                                                     
+virtual Status Authenticate(const std::string& user, const std::string& passwd) = 0;
+virtual Status Authenticate(const std::string& secret) = 0;
+/// The object storage optimized memory copy.
+virtual void MemoryCopy(void* dest, const void* src, size_t len) = 0;
```

Comment:
Can you say more about how this would get used?

Reply:
Currently memcpy() is used for data copy to object data buffer in writing object chunks (ObjectBufferPool::WriteChunk)
For objects with data buffer on CXL shared memory between hosts, additional optimization may be needed in addition to memcpy() to guarantee the cache coherence so that other hosts can directly access the shared object buffer directly. The plugin can provide that functionality.